#BiocManager::install("ggimage")

library(ChIPseeker)

library(TxDb.Hsapiens.UCSC.hg38.knownGene)

library(GenomicAlignments)
library(GenomicFeatures)

library(clusterProfiler)
library(biomaRt)
library(org.Hs.eg.db)
library(ReactomePA)
library(ggupset)
library(ggimage)
library(doParallel)
cl <- makePSOCKcluster(8)
registerDoParallel(cl)

######
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
### the peak file generated by bedtools subtract
pth2peaks_bed="/Users/guo/Downloads/th17_special.bed"

peaks.bed=read.table(pth2peaks_bed, sep="\t", header=FALSE, blank.lines.skip=TRUE)
rownames(peaks.bed)=peaks.bed[,4]
### construct GRanges object
peaks.gr <- GRanges(seqnames=peaks.bed[,1], ranges=IRanges(peaks.bed[,2], peaks.bed[,3]), strand="*", mcols=data.frame(peakID=peaks.bed[,4]))

peaks.gr

covplot(peaks.gr, chrs=c("chr14", "chr15"))

###
bed.annot = annotatePeak(peaks.gr, tssRegion=c(-3000, 3000),TxDb=txdb, annoDb="org.Hs.eg.db")
bed.annot

annot_peaks=as.data.frame(bed.annot)

head(annot_peaks)
### genomic region
upsetplot(bed.annot, vennpie=TRUE)
###
plotDistToTSS(bed.annot, title="Distribution of ATAC-seq peaks loci\nrelative to TSS")

# pathway anno by cluterProfile 
pathway.reac <- enrichPathway(as.data.frame(annot_peaks)$geneId)

head(pathway.reac)
colnames(as.data.frame(pathway.reac))
pathway.reac[1:10,c(1:7,9)]
barplot(pathway.reac, showCategory=20) 
dotplot(pathway.reac, showCategory=20) 
###
pathway.GO <- enrichGO(as.data.frame(annot_peaks)$geneId, org.Hs.eg.db, ont = "MF")
pathway.GO[1:11,c(1,2,7)]
barplot(pathway.GO, showCategory=20) 
#####


#BiocManager::install("ATACseqQC")

#BiocManager::install("MotifDb")

#BiocManager::install("BSgenome.Hsapiens.UCSC.hg38")
##################################################################
library(ATACseqQC)
library(MotifDb)

library(BSgenome.Hsapiens.UCSC.hg38)
genome <- Hsapiens

shifted.bamFile="/Users/guo/Downloads/SRR5831513.sorted.coordinate.bam"

#  analysis  chr14
seqlev <- "chr14"
seqlev <-  paste0("chr", c(1:22, "X", "Y"))
########################################################################v

CTCF <- query(MotifDb, c("CTCF"))
CTCF <- as.list(CTCF)
print(CTCF[[1]], digits=2)



ctcf <- factorFootprints(shifted.bamFile, pfm=CTCF[[1]],
                         genome=genome,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)


ctcf$spearman.correlation
sigs <- factorFootprints(shifted.bamFile, pfm=CTCF[[1]],
                         genome=genome,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)
########################################################################

RFX5 <- query(MotifDb, c("RFX5"))
RFX5 <- as.list(RFX5)


rfx5 <- factorFootprints(shifted.bamFile, pfm=RFX5[[1]],
                         genome=genome,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)

rfx5$spearman.correlation$`+`$estimate
rfx5$spearman.correlation$`+`$p.value


pdf("rfx5_footprnt.pdf")
rfx5 <- factorFootprints(shifted.bamFile, pfm=RFX5[[1]],
                         genome=genome,
                         min.score="90%", seqlev=seqlev,
                         upstream=100, downstream=100)
dev.off()
########################################################################v
STAT3 <- query(MotifDb, c("STAT3"))
STAT3 <- as.list(STAT3)


stat3 <- factorFootprints(shifted.bamFile, pfm=STAT3[[1]],
                          genome=genome,
                          min.score="90%", seqlev=paste0("chr", c(1:22, "X", "Y")),
                          upstream=100, downstream=100)

stat3$spearman.correlation$`+`$estimate
stat3$spearman.correlation$`+`$p.value

stat3$Profile.segmentation

pdf("stat3_footprnt.pdf")
stat3 <- factorFootprints(shifted.bamFile, pfm=STAT3[[1]],
                          genome=genome,
                          min.score="90%", seqlev=seqlev,
                          upstream=100, downstream=100)
dev.off()

########################


